ctapipe/tools/dl1_merge.py

```diff
@@ -0,0 +1,359 @@
+    flags = {
+        "skip-images": (
+            {"MergeTool": {"skip_images": True}},
+            "Skip DL1/Event/Telescope and Simulation/Event/Telescope images in output",
+        ),
+        "skip-simu-images": (
+            {"MergeTool": {"skip_simu_images": True}},
+            "Skip Simulation/Event/Telescope images in output",
+        ),
+        "skip-parameters": (
+            {"MergeTool": {"skip_parameters": True}},
+            "Skip DL1/Event/Telescope parameters in output",
+        ),
+        "overwrite": (
+            {"MergeTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"MergeTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        ),
+    }
+
+    def setup(self):
+        # prepare output path:
+        self.output_path = self.output_path.expanduser()
+        if self.output_path.exists():
+            if self.overwrite:
+                self.log.warning(f"Overwriting {self.output_path}")
+                self.output_path.unlink()
+            else:
+                self.log.critical(
+                    f"Output file {self.output_path} exists, "
+                    "use `--overwrite` to overwrite"
+                )
+                sys.exit(1)
+
+        PROV.add_output_file(str(self.output_path))
+
+        if self.skip_parameters is True and self.skip_images is True:
+            self.log.warning("Skip-parameters and skip-images are both" "set to True")
+
+        # Get input Files
+        args = self.parser.parse_args(self.extra_args)
+        self.input_files = args.input_files
+        if self.input_dir is not None:
+            self.input_files.extend(sorted(self.input_dir.glob(self.file_pattern)))
+        if not self.input_files:
+            self.input_files = sorted(Path(".").glob(self.file_pattern))
+
+        # create output file with subarray from first file
+        self.first_subarray = SubarrayDescription.from_hdf(self.input_files[0])
+        self.first_subarray.to_hdf(self.output_path)
+        self.output_file = tables.open_file(self.output_path, mode="a")
+
+    def check_file(self, file):
+        # Check that the file is not broken or any node is missing
+        file_path = file.root._v_file.filename
+        current_subarray = SubarrayDescription.from_hdf(file_path)
+        # Check subarrays
+        if self.first_subarray != current_subarray:
+            self.log.warning(f"Subarray does not match for {file_path}. " "Skip File.")
+            self.skip_file = True
+
+        # Gives warning if tables for listed nodes in 'optional_nodes'
+        # are missing but continues merging the rest of the file. If any
+        # other node from 'required_nodes' is missing, except the given combinations
+        # of traits and blocklist, the whole file will be skipped.
+        for node in required_nodes:
+            if self.is_simu is False and node in allowlist_simu:
```

Comment:
It might be better to build the allowlists / required lists / ... once in `setup` according to the provided options and then work with single sets in this check here.

Reply:
Also, membership checks are better done with sets than lists.